packages/rs-sdk/src/core/transaction.rs

```diff
@@ -65,11 +63,30 @@ impl Sdk {
 
     /// Waits for a response for the asset lock proof
     pub async fn wait_for_asset_lock_proof_for_transaction(
+        &self,
         mut stream: dapi_grpc::tonic::Streaming<TransactionsWithProofsResponse>,
         transaction: &Transaction,
-        time_out_ms: Option<u64>,
-        // core_chain_locked_height: u32,
+        time_out: Option<Duration>,
     ) -> Result<AssetLockProof, Error> {
+        // Make sure that the streaming timeout is greater than processing timeout
+        // TODO: We should with actual request timeout but not default one. Fix it during refactoring
+        if let (
+            RequestSettings {
+                timeout: Some(streaming_timeout),
+                ..
+            },
+            Some(processing_timeout),
+        ) = (TransactionsWithProofsRequest::SETTINGS_OVERRIDES, time_out)
```

Comment:
Please double-check the whole settings overriding logic in rs-dapi-client, it does some merging etc.